internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -94,12 +101,41 @@ func parsePortMapping(s *string) (port *string, protocol *string, err error) {
 	}
 }
 
+// convertCapacityProviders transforms the manifest fields into a format
+// parsable by the templates pkg.
+func convertCapacityProviders(a *manifest.Autoscaling) ([]*template.CapacityProviderStrategy, error) {
```

Comment:
Since both `convertCapacityProviders` and `convertAutoscaling` take the same parameter, does it make more sense to combine them to just `convertAutoscaling` and return a struct with all info for the field in the manifest? It's a little strange to see a manifest field is parsed into three different fields. Ideally I think we should handle the `DesiredCountOnSpot` in the `TaskCount` in parameters, instead of using it in go template.

Reply:
Yeah I totally get that it's weird! Unfortunately I wasn't sure how else to separate the arguments in a logical way, since range and count fields end up being part of the conversion logic. I also didn't feel like coupling the output data structure seemed like the right move, especially since `template.AutoscalingOpts` and Capacity  Provider Strategies aren't always used together and it seemed like having to handle nil checks every time for the combined data structure would result in more maintenance burden down the road.

Reply:
Efe's proposal looks good to me https://github.com/aws/copilot-cli/pull/2188/files#r617104857. So that we'll just have `convertAdvancedCount` returning to the new struct that I mentioned before.

Reply:
Fix'd! https://github.com/aws/copilot-cli/pull/2188/commits/bc88a31ff4b5cb5cd504c0ebf8e7b42bd17f48b4